python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -476,6 +476,10 @@ MlirModule synthesizeKernel(const std::string &name, MlirModule module,
   PassManager pm(context);
   pm.addPass(cudaq::opt::createQuakeSynthesizer(name, rawArgs));
   pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
+  pm.addPass(cudaq::opt::createLiftArrayAllocPass());
+  pm.addPass(cudaq::opt::createStatePreparation(name));
```

Comment:
I think that this code implies that this state decomposition pass will be run for any Python kernel with qvector init from state, even for simulators. But I would think we only want to run this when we are targeting physical QPUs. For simulators we can just keep forwarding the data like we've done before.

Reply:
I will try to figure out what the backend is and add a check

Reply:
Alternatively, you can add the pass in the hardware configuration file, like `runtime/cudaq/platform/default/rest/helpers/quantinuum/quantinuum.config`

Reply:
Added condition to avoid running state prep for simulators.

Reply:
>Alternatively, you can add the pass in the hardware configuration file, like 

I have this too, but this API is just running synthesis without loading the pipelines from the config files. So added a condition for now.